### PR TITLE
Add simple Maven Plugin

### PR DESCRIPTION
**Pull request by jgallimore**
We discussed this a bit on the old repository, so here it is, rebased against the new Maven build. See https://github.com/tbitonti/jakartaee-prototype/pull/102.

The Mojo provides a classifier parameter, and any attached artifacts get transformed, and the transformed artifacts are attached with whatever value is specified appended to the artifacts classifier.

For example, in the TomEE build we have the following plugin config:
```
      <plugin>
        <groupId>org.eclipse.transformer</groupId>
        <artifactId>org.eclipse.transformer.mave
        <version>0.1.1-SNAPSHOT</version>
        <configuration>
          <classifier>jakartaee9</classifier>
        </configuration>
        <executions>
          <execution>
            <goals>
              <goal>run</goal>
            </goals>
            <phase>package</phase>
          </execution>
        </executions>
      </plugin>
```

And the module has the following attached artifacts:

org.apache.tomee:apache-tomee:webprofile:8.0.3-SNAPSHOT:zip
org.apache.tomee:apache-tomee:microprofile:8.0.3-SNAPSHOT:zip
org.apache.tomee:apache-tomee:plume:8.0.3-SNAPSHOT:zip
org.apache.tomee:apache-tomee:plus:8.0.3-SNAPSHOT:zip

Once the plugin is run, the following additional artifacts are attached:

org.apache.tomee:apache-tomee:webprofile-jakartaee9:8.0.3-SNAPSHOT:zip
org.apache.tomee:apache-tomee:microprofile-jakartaee9:8.0.3-SNAPSHOT:zip
org.apache.tomee:apache-tomee:plume-jakartaee9:8.0.3-SNAPSHOT:zip
org.apache.tomee:apache-tomee:plus-jakartaee9:8.0.3-SNAPSHOT:zip

### CODE REVIEW

**Comment by jgallimore** [comment]
This relates to issue #48.

File: org.eclipse.transformer.maven/pom.xml [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+		<dependency>
+			<groupId>org.assertj</groupId>
+			<artifactId>assertj-core</artifactId>
+		</dependency>
+
+        <dependency>
+            <groupId>org.jboss.shrinkwrap</groupId>
+            <artifactId>shrinkwrap-api</artifactId>
+            <version>1.2.6</version>
+            <scope>compile</scope>
```
Comment (bjhargrave) [inline org.eclipse.transformer.maven/pom.xml:137]: I think this is just a test dependency, right?

File: org.eclipse.transformer.maven/pom.xml [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+                </executions>
+                <configuration>
+                    <mojoDependencies>
+                    </mojoDependencies>
+                </configuration>
+            </plugin>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-jar-plugin</artifactId>
+				<version>3.2.0</version>
```
Comment (bjhargrave) [inline org.eclipse.transformer.maven/pom.xml:182]: We already set the version in the parent pom's pluginManagement.

File: org.eclipse.transformer.maven/pom.xml [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+				<configuration>
+					<archive  combine.self="override"/>
+				</configuration>
+			</plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-release-plugin</artifactId>
```
Comment (bjhargrave) [inline org.eclipse.transformer.maven/pom.xml:189]: I don't think we need this plugin.

**Comment by jgallimore** [comment]
I'll recheck in the morning, but I think I have managed to address all the feedback. Thanks for the review!

File: org.eclipse.transformer.maven/src/main/java/org/eclipse/transformer/maven/TransformMojo.java [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+        final String targetClassifier = (sourceClassifier == null || sourceClassifier.length() == 0) ?
+            this.classifier : sourceClassifier + "-" + this.classifier;
+
+        final File targetFile = new File(outputDirectory,
+            sourceArtifact.getArtifactId() + "-" +
+                targetClassifier + "-" + sourceArtifact.getVersion() + "."
+                + sourceArtifact.getType());
+
+        transformer.setArgs(new String[] { sourceArtifact.getFile().getAbsolutePath(), targetFile.getAbsolutePath() });
+        int rc = transformer.run();
```
[the remark this answers is not in the document]
Comment (jgallimore) [reply org.eclipse.transformer.maven/src/main/java/org/eclipse/transformer/maven/TransformMojo.java:114]: This should now fail the build with the correct description.

**Comment by jgallimore** [comment]
It looks like the feedback here is addressed. I've rebased and resolved the conflicts. Would it be possible for someone to have another look?

**Pull request merged**